src/analysis/inner_product_space/finite_dimensional.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2023 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import analysis.inner_product_space.adjoint
+
+/-!
+# Finite-dimensional inner product spaces
+
+In this file, we prove some results in finite-dimensional inner product spaces.
+
+## Notation
+
+This file uses the local notation `P _` for `orthogonal_projection _`
+and `P _ᗮ` for `orthogonal_projection _ᗮ`.
+
+We let `V` be a finite-dimensional inner product space over `ℂ`.
+-/
+
+variables {V : Type*} [inner_product_space ℂ V]
+local notation `P`U := (orthogonal_projection U)
+local notation `P`U`ᗮ` := (orthogonal_projection Uᗮ)
+
+/-- `U` is `T` invariant: `∀ u : V`, if `u ∈ U` then `T u ∈ U`-/
+def invariant_subspace (U : submodule ℂ V) (T : V →L[ℂ] V) :
+ Prop := (∀ u : V, u ∈ U → T u ∈ U)
```

Comment:
I think `U ≤ U.comap T` would be a better definition here; I'm pretty sure they're definitionally equal

Reply:
Done.